openpype/hosts/cinema4d/plugins/publish/collect_instances.py

```diff
@@ -0,0 +1,185 @@
+            append this input data into the instance, as we do
+            for `pype.rig` with collect_history.
+
+    """
+
+    label = "Collect Instances"
+    order = pyblish.api.CollectorOrder
+    hosts = ["cinema4d"]
+
+    def process(self, context, doc=None):
+        if not doc:
+            doc = c4d.documents.GetActiveDocument()
+
+        objectset = set()
+        for obj in lib.walk_hierarchy(doc.GetFirstObject()):
+            if obj.attrs.get("id"):
+                objectset.add(obj.attrs)
+
+
+        ctx_frame_start = context.data['frameStart']
+        ctx_frame_end = context.data['frameEnd']
+        ctx_handle_start = context.data['handleStart']
+        ctx_handle_end = context.data['handleEnd']
+        ctx_frame_start_handle = context.data['frameStartHandle']
+        ctx_frame_end_handle = context.data['frameEndHandle']
+
+        context.data['objectsets'] = objectset
+        for objset in objectset:
+
+            if not objset.get("id"):
+                continue
+
+            if objset.get("id") != "pyblish.avalon.instance":
+                continue
+
+            # The developer is responsible for specifying
+            # the family of each instance.
+            has_family = objset.get("family")
+            assert has_family, "\"%s\" was missing a family" % objset
+
+            members = objset.get("SELECTIONOBJECT_LIST")
+            if members is None:
+                self.log.warning("Skipped empty instance: \"%s\" " % objset)
+                continue
+
+            self.log.info("Creating instance for {}".format(lib.serialize_c4d_data(objset.op)))
+
+            data = dict()
+
+            for key, value in objset:
+                data[key] = lib.serialize_c4d_data(value)
+
+            # temporarily translation of `active` to `publish` till issue has
+            # been resolved, https://github.com/pyblish/pyblish-base/issues/307
+            if "active" in data:
+                data["publish"] = data["active"]
+
+            # Collect members - c4d.BaseObject
+            members = [lib.ObjectPath(obj=members.ObjectFromIndex(doc, idx)) for idx in range(members.GetObjectCount())]
+
+
+            # Collect Children - c4d.BaseObject
+            children = [lib.ObjectPath(obj=obj) for member in members for obj in lib.walk_hierarchy(member.obj.GetDown())]
+
+            parents = []
+            if data.get("includeParentHierarchy", True):
+                # If `includeParentHierarchy` then include the parents
+                # so they will also be picked up in the instance by validators
+                parents = [lib.ObjectPath(obj=obj) for obj in self.get_all_parents(members)]
+            members_hierarchy = list(
+                    set(
+                        [str(x) for x in members] + \
+                        [str(x) for x in children] + \
+                        [str(x) for x in parents]
+                    )
+                )
```

Comment:
```suggestion
            members_hierarchy = list(
                    set(str(x) for x in members + children + parents))
                )
```
Would this be more readable?